resources/assets/js/volumes/cloneForm.vue

```diff
@@ -0,0 +1,190 @@
+        },
+        cannotSubmit() {
+            return this.name === '' || this.selectedProjectId < 0 || this.loading;
+        },
+        getCloneUrl() {
+            return this.cloneUrlTemplate.replace(':pid', this.selectedProjectId);
+        },
+    },
+    methods: {
+        setProject(project) {
+            this.selectedProjectId = project.id;
+        },
+        async loadFilesMatchingPattern() {
+            if (this.filePattern.length > 0) {
+                this.startLoading();
+                let id2filenames = await VolumeApi.queryFilenames({id: this.id})
+                    .then((response) => {
+                        return response.body;
+                    }, handleErrorResponse);
+                VolumeApi.queryFilesWithFilename({id: this.id, pattern: this.filePattern})
+                    .then((response2) => {
+                        let ids = response2.body;
+                        this.setMatchedFiles(ids.map(id => {
+                            return {id: id, filename: id2filenames[id]}
+                        }));
+                    }, handleErrorResponse)
+                    .finally(this.finishLoading);
+            }
+        },
+        setMatchedFiles(filenames) {
+            this.selectedFiles = filenames;
+        },
+        flatLabels(trees) {
+            let labels = [];
+            trees.forEach(function (tree) {
+                Array.prototype.push.apply(labels, tree.labels);
+            });
+
+            return labels;
+        },
+        setFileLabels(labelIds) {
+            this.fileLabelIds = labelIds;
+        },
+        setAnnotationLabels(labelIds) {
+            this.annotationLabelIds = labelIds;
+        },
+        initializeLabelTrees(fileLabelTrees, annotationLabelTrees) {
+            const nbrTrees = fileLabelTrees.length;  // labels of fileLabelTrees and annotationLabelTrees are equal
+
+            if (this.fileLabelIds.length > 0) {
+                this.cloneFileLabels = true;
+                this.restrictFileLabels = true;
+            }
+
+            if (this.annotationLabelIds.length > 0) {
+                this.cloneAnnotationLabels = true;
+                this.restrictAnnotationLabels = true;
+            }
+
+            for (let i = 0; i < nbrTrees; i++) {
+                fileLabelTrees[i].labels.forEach((label) => this.fileLabelIds.includes(String(label.id)) ?
+                    label.selected = true : label.selected = false);
+                annotationLabelTrees[i].labels.forEach((label) => this.annotationLabelIds.includes(String(label.id)) ?
+                    label.selected = true : label.selected = false);
+            }
+
+            this.fileLabelTrees = fileLabelTrees;
+            this.annotationLabelTrees = annotationLabelTrees;
+        },
+        initializeFileList(ids, filenames) {
+            const nbrFiles = ids.length;
+            if (nbrFiles > 0) {
+                this.cloneFiles = true;
```

Comment:
This is also the same with the file labels and annotation labels above. If I select "filter images/videos", then add a filter query and get a file list and then uncheck "filter images/videos" again, the file list is still there as hidden form fields.